lib/bitcoin/BitcoinClient.ts

```diff
@@ -96,46 +150,39 @@ export default class BitcoinClient implements IBitcoinClient {
     return response;
   }
 
-  public async getRawTransaction (transactionId: string): Promise<Transaction> {
-    const request = {
-      method: 'getrawtransaction',
-      params: [
-        transactionId,  // transaction id
-        0   // get the raw hex-encoded string
-      ]
-    };
+  public async getBalanceInSatoshis (): Promise<number> {
 
-    const hexEncodedTransaction = await this.rpcCall(request, true);
-    const transactionBuffer = Buffer.from(hexEncodedTransaction, 'hex');
+    const unspentOutputs = await this.getUnspentOutputs(this.privateKeyAddress);
+
+    const unspentSatoshis = unspentOutputs.reduce((total, unspentOutput) => {
+      return total + unspentOutput.satoshis;
+    }, 0);
 
-    return BitcoinClient.createTransactionFromBuffer(transactionBuffer);
+    return unspentSatoshis;
   }
 
-  public async getUnspentCoins (address: Address): Promise<Transaction.UnspentOutput[]> {
+  public async getTransactionFeeInSatoshis (transactionId: string): Promise<number> {
 
-    // Retrieve all transactions by addressToSearch via BCoin Node API /tx/address/$address endpoint
-    const addressToSearch = address.toString();
-    console.info(`Getting unspent coins for ${addressToSearch}`);
-    const request = {
-      method: 'listunspent',
-      params: [
-        null,
-        null,
-        [addressToSearch]
-      ]
-    };
-    const response: Array<any> = await this.rpcCall(request, true);
+    const transaction = await this.getRawTransaction(transactionId);
 
-    const unspentTransactions = response.map((coin) => {
-      return new Transaction.UnspentOutput(coin);
-    });
+    let inputSatoshiSum = 0;
+    for (let i = 0 ; i < transaction.inputs.length ; i++) {
 
-    console.info(`Returning ${unspentTransactions.length} coins`);
+      const currentInput = transaction.inputs[i];
+      const transactionOutValue = await this.getTransactionOutValueInSatoshi(currentInput.previousTransactionId, currentInput.outputIndexInPreviousTransaction);
 
-    return unspentTransactions;
+      inputSatoshiSum += transactionOutValue;
+    }
+
+    // transaction outputs in satoshis
+    const transactionOutputs: number[] = transaction.outputs.map((output) => output.satoshis);
+
+    const outputSatoshiSum = transactionOutputs.reduce((sum, value) => sum + value, 0);

```

Comment:
outputs.map(...).reduce(...); ?

Reply:
Feel like this is more readable in separate lines so will keep it.